packages/dhis2/src/tracker.js

```diff
@@ -0,0 +1,136 @@
+import { expandReferences } from '@openfn/language-common/util';
+import * as util from './Utils';
+
+/**
+ * State object
+ * @typedef {Object} Dhis2State
+ * @private
+ * @property data - The response body (as JSON)
+ * @property response - The HTTP response from the Dhis2 server (excluding the body)
+ * @property references - An array of all previous data objects used in the Job
+ */
+
+/**
+ * All options, apart from those listed here, will be appended as query parameters to the URL
+ * @typedef {Object} TrackerOptions
+ * @property {string} [parseAs='json'] - The response format to parse (e.g., 'json', 'text', 'stream', or 'base64'. Defaults to `json`
+ * @property {string} [apiVersion=42] - The apiVersion of the request. Defaults to 42.
+ */
+
+/**
+ * Import data into DHIS2 using the tracker endpoint.
+ * @alias import
+ * @public
+ * @function
+ * @param {string} strategy - The effect the import should have. Can either be CREATE, UPDATE, CREATE_AND_UPDATE and DELETE.
+ * @param {object} payload - The data to be imported.
+ * @param {TrackerOptions} [options] - An optional object containing parseAs, and apiVersion, and queries for the request
+ * @state {Dhis2State}
+ * @returns {Operation}
+ * @example <caption>Import some data and pass the `atomicMode` parameter</caption>
+ * tracker.import('CREATE', $.trackerData, { atomicMode: 'ALL' })
+ * @example <caption>Import a trackedEntity resource</caption>
+ * tracker.import(
+ *   'CREATE',
+ *   {
+ *     trackedEntities: [
+ *       {
+ *         orgUnit: 'TSyzvBiovKh',
+ *         trackedEntityType: 'nEenWmSyUEp',
+ *         attributes: [
+ *           {
+ *             attribute: 'w75KJ2mc4zz',
+ *             value: 'Gigiwe',
+ *           },
+ *         ],
+ *       },
+ *     ],
+ *   },
+ *   {
+ *     atomicMode: 'ALL',
+ *   }
+ * );
+ */
+function _import(strategy, payload, options = {}) {
+  return async state => {
+    console.log('Preparing tracker import operation...');
+
+    const [resolvedStrategy, resolvedPayload, resolvedOptions] =
+      expandReferences(state, strategy, payload, options);
+
+    const { apiVersion, parseAs, ...query } = resolvedOptions;
+
+    const response = await util.request(state.configuration, {
+      method: 'POST',
+      path: util.prefixVersionToPath(
+        state.configuration,
+        {
+          ...resolvedOptions,
+          resolvedStrategy,
+        },
+        'tracker'
+      ),
+      options: {
+        apiVersion,
```

Comment:
@hunterachieng I don't think `apiVersion` should be passed into the request options like this? It's not used, and so it's really confusing to pass it here.